src/torrent.cpp

```diff
@@ -40,6 +41,15 @@ TorrentFile::TorrentFile(const BencodeValue& tree) {
 
   auto pieces =
       dynamic_cast<BencodeString&>(*info_dict.at("pieces")).value();
-  this->piece_hashes = hash::split_piece_hashes(pieces);
+
+  auto r_hashes = hash::split_piece_hashes(pieces);
+
+  if(!r_hashes.valid()){
+      auto logger = spdlog::get("custom");
+      logger->error("Could not split piece hashes {}",
```

Comment:
Doesn't this print something like `Could not split piece hashes 0`? I'd prefer to have the enum printed as a string rather than an integer.